src/services/create-project.service.test.js

```diff
@@ -57,3 +108,86 @@ describe('getBuildInstructions', () => {
     );
   });
 });
+
+describe('CreateProject', () => {
+  const newProject = {
+    projectName: 'Awesome project',
+    projectType: 'create-react-app',
+    projectIcon: 'Icon',
+    ProjectStarter: null,
+  };
+  let mockStatusUpdate;
+  let mockErrorHandler;
+  let mockCompleteHandler;
+  let callParams;
+
+  beforeEach(() => {
+    global.console.error = jest.fn();
+
+    mockStatusUpdate = jest.fn();
+    mockErrorHandler = jest.fn();
+    mockCompleteHandler = jest.fn();
+
+    callParams = [
+      newProject,
+      'projectHomePath/',
+      mockStatusUpdate,
+      mockErrorHandler,
+      mockCompleteHandler,
+    ];
+    DISABLE.status = false;
+  });
+
+  afterAll(() => {
+    jest.unmock(global.console.error);
+  });
+
+  it('should create a fake project for debugging', () => {
+    DISABLE.status = true;
+    createProjectSvc(...callParams);
+
+    expect(mockCompleteHandler).toHaveBeenCalledWith(FAKE_CRA_PROJECT);
+  });
+
+  it('should create home directory if it does not exist', () => {
+    createProjectSvc(...callParams);
+
+    expect(fs.existsSync).toHaveBeenLastCalledWith('projectHomePath/');
+    expect(fs.mkdirSync).toHaveBeenLastCalledWith('projectHomePath/');
+    expect(mockStatusUpdate).toHaveBeenLastCalledWith(
+      expect.stringMatching(/created parent directory/i)
+    );
+  });
+
+  it('should listen to process stdout & stderr', () => {
+    createProjectSvc(...callParams);
+
+    eventMap.stdout.data('some data');
+    expect(mockStatusUpdate).toHaveBeenCalledWith('some data');
+
+    eventMap.stderr.data('an error');
+    expect(mockErrorHandler).toHaveBeenCalledWith('an error');
+  });
+
+  it('should listen to process close event', () => {
+    createProjectSvc(...callParams);
+
+    eventMap.close();
+
+    expect(mockStatusUpdate).toHaveBeenLastCalledWith(
+      expect.stringMatching(/dependencies installed/i)
+    );
+    // read failure
+    expect(global.console.error).toHaveBeenCalled();
+
+    // new close event & test onComplete
+    eventMap.close();
+    expect(mockCompleteHandler).toHaveBeenCalled();
+  });
+
+  it('should add a commit for create-react-app', () => {
+    createProjectSvc(...callParams);
+    eventMap.close();
+    expect(childProcess.exec).toHaveBeenCalled();
```

Comment:
@superhawk610 should we be more detailed here? I think just checking that it is called is not enough as the complete command could be changed with-out a failing test.
Or is it OK to just check that it is called? It could be OK as we're having no benefit if we completely checking the call - it would be just a failing test that we would have to update if there would be a change in the command.

Reply:
I think a small args check would be nice here, something like

```js
    expect(childProcess.exec).toHaveBeenCalledWith(
      expect.stringMatching('Add Guppy data to package.json'),
      { cwd: 'projectHomePath/awesome-project' }
    );
```

Reply:
Good idea, I've added that.
 
I mocked `path.join` as Windows used backslashes and I'd like to be consistent with forward-slash.